resources/conditions/error_condition.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package conditions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/cloudbeat/resources/fetching"
+)
+
+type ErrorCondition struct {
```

Comment:
As far as I understand, this struct is mainly here to write the same error log over and over again.
First, I think the first error log that you have added (that happen once) should be enough.
Second, in the near future, we would like to deprecate the conditions and instead use `Audotdiscovery` feature in beats. Already opened a ticket for that [fetchers config based on autodiscover](https://github.com/elastic/security-team/issues/3334).

As for this PR, I would avoid of adding a new condition at the moment. (Maybe we should already mark this class as deprecated?)
WFYT?

Reply:
Another option I can think of that makes `ErrorCondition` redundant:
`getConditions` should return the error up.
In case of an error, the fetcher should not be registered at all.
Currently, we only have a single fetcher condition and its requirement fits into what I described.